src/internal/_concat.js

```diff
@@ -0,0 +1,45 @@
+import { Iterable } from 'immutable';
+import { concat } from './TransmuteCollection';
+
+const checkSubject = subject => {
+  if (subject === null || subject === undefined) {
+    throw new Error('Cannot call concat on `null` or `undefined`');
```

Comment:
If you change the `protocol.TYPE` in args like I suggested above, you may be able to do away with this `checkSubject` util entirely. Once the protocol is choosing an implementation based on subject's type, if there's no implementation for `null` or `undefined` the protocol will [throw a unimplemented exception for you](https://github.com/HubSpot/transmute/blob/master/src/protocol.js#L68).